include/xsimd/math/xsimd_estrin.hpp

```diff
@@ -0,0 +1,103 @@
+/***************************************************************************
+* Copyright (c) Johan Mabille, Sylvain Corlay, Wolf Vollprecht and         *
+* Martin Renou                                                             *
+* Copyright (c) QuantStack                                                 *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XSIMD_ESTRIN_HPP
+#define XSIMD_ESTRIN_HPP
+
+#include "../types/xsimd_types_include.hpp"
+
+namespace xsimd
+{
+    namespace detail
+    {
+        template <class T, uint64_t c, typename T::value_type* = nullptr>
+        inline T coef() noexcept
```

Comment:
Why not:
```cpp
std::enable_if<!std::is_scalar<T>::value, T>::type coef() noexcept
```

to keep symmetry with the other one?

Reply:
Not too familiar with SFINAE but I wasn't able to make this work with float and double (which is why I am guarding the benchmarks with the XSIMD_POLY_BENCHMARKS define). The problem is that fma is not found because no ADL is performed and if I include xsimd_scalar.hpp, that needs xsimd_power.hpp which needs xsimd_horner.hpp.